src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,86 @@
+package duke.task;
+
+import duke.exception.InvalidRequestException;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a TaskList and consists of information and methods related to the list of tasks.
+ */
+public class TaskList {
+
+    private ArrayList<Task> listOfTasks;
+
+    /**
+     * Constructs a TaskList.
+     *
+     * @param listOfTasks The list of tasks.
+     */
+    public TaskList(ArrayList<Task> listOfTasks) {
+        this.listOfTasks = listOfTasks;
+    }
+
+    /**
+     * Adds a duke.task to the list of tasks.
+     *
+     * @param newTask The duke.task to be added.
+     */
+    public void addTask(Task newTask) {
+
+        listOfTasks.add(newTask);
+    }
+
+    /**
+     * Deletes a duke.task from the list of tasks.
+     *
+     * @param index The index of the duke.task to be deleted.
+     * @throws InvalidRequestException If the command is invalid.
+     */
+    public void deleteTask(int index) throws InvalidRequestException {
+            if (listOfTasks.size() < index || index < 0) {
```

Comment:
There are some issues with the indentation in this class. For example,  the expected number of spaces should be 8 instead of 12 in line 40.

Reply:
Yes, I have changed that. Thank you for pointing that out!